src/web/topic/components/TopicPane/TopicPane.tsx

```diff
@@ -1,86 +1,96 @@
+
+    const unbindView = emitter.on("viewTopicDetails", () => {
+      setSelectedTab("Details");
+      setIsOpen(true);
+    });
+
+    return () => {
+      unbindView();
+    };
+  }, [tabs]);
+
   const handlePaneToggle = () => {
-    if (isTopicPaneOpen) {
-      setIsTopicPaneOpen(false);
-    } else {
-      setIsTopicPaneOpen(true);
-    }
+    setIsOpen(!isOpen);
+  };
+  const handleTabChange = (_: React.SyntheticEvent, value: TopicTabName) => {
+    setSelectedTab(value);
   };
 
-  const ToggleIcon = isTopicPaneOpen
-    ? isLandscape
-      ? ChevronLeft
-      : KeyboardArrowDown
-    : AutoStories;
-
-  return (
-    <>
-      {/* div to enable menu button to be positioned to the right of the pane */}
-      <PositionedDiv>
-        <TogglePaneButton onClick={handlePaneToggle} color="primary" isLandscape={isLandscape}>
-          <ToggleIcon />
-        </TogglePaneButton>
-        {/* `permanent` because `persistent` adds transitions that conflict with our styles */}
-        <StyledDrawer
-          variant="permanent"
-          open={isTopicPaneOpen}
-          anchor={isLandscape ? "left" : "bottom"}
-          isLandscape={isLandscape}
-        >
-          <TabContext value={selectedTab}>
-            <TabList
-              onChange={(_, newValue: "Details" | "Views") => setSelectedTab(newValue)}
-              centered
-            >
-              <Tab label="Details" value="Details" />
-              <Tab label="Views" value="Views" />
-            </TabList>
+  const ToggleIcon = isOpen ? ANCHOR_ICON_MAP[anchor] : AutoStories;
 
-            <TabPanel value="Details">
-              {selectedGraphPart !== null ? (
-                // Key ensures that details components re-render, re-setting default values.
-                // Could consider using a `viewingGraphPart` separate from selected so that graph
-                // part details for parts outside of the shown diagram can be displayed, without
-                // losing tracking of the selected part per diagram.
-                <GraphPartDetails graphPart={selectedGraphPart} key={selectedGraphPart.id} />
-              ) : (
-                <TopicDetails />
-              )}
-            </TabPanel>
+  const TabPanelContent = {
+    Details:
+      selectedGraphPart !== null ? (
+        <GraphPartDetails graphPart={selectedGraphPart} key={selectedGraphPart.id} />
+      ) : (
+        <TopicDetails />
+      ),
+    Views: <TopicViews />,
+  };
 
-            <TabPanel value="Views">
-              <TopicViews />
+  if (tabs.length == 0) return <></>;
+  return (
+    <PositionedDiv>
+      <TogglePaneButton onClick={handlePaneToggle} color="primary" anchor={anchor}>
+        <ToggleIcon />
+      </TogglePaneButton>
+      <StyledDrawer variant="permanent" open={isOpen} anchor={anchor}>
+        <TabContext value={selectedTab}>
+          <TabList onChange={handleTabChange} centered>
+            {tabs.map((name, idx) => (
```

Comment:
```suggestion
            {tabs.map((tab, idx) => (
```

np but it's a little odd that the array `tabs` is containing `name`s instead of `tab`s